### PR TITLE
Adding logic for undodelete

### PR DESCRIPTION
**Pull request by InnaMel**

### CODE REVIEW

File: DuplicatesFinder_v4/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -151,7 +153,7 @@ public ICommand OnClickExport
             {
                 return onClickExport ?? (onClickExport = new RelayCommand((r) =>
                 {
-                    Task.Run(() => DuplicatesViewModel.SaveToTxt());
+                    Task.Run(() => DuplicatesViewModel.SaveToTxtAsync());
```
Comment (YuriZakitny) [inline DuplicatesFinder_v4/ViewModels/MainViewModel.cs:156]: Incorrect logic.
You wan to start and not wait async method.
In this case you have two options:
1. Use Task.Run to execute **non-async** method
2. Just call async method without waiting (without call of await)

File: DuplicatesFinder_v4/ViewModels/MainViewModel.cs [hunk 2]
```diff
@@ -151,7 +153,7 @@ public ICommand OnClickExport
             {
                 return onClickExport ?? (onClickExport = new RelayCommand((r) =>
                 {
-                    Task.Run(() => DuplicatesViewModel.SaveToTxt());
+                    Task.Run(() => DuplicatesViewModel.SaveToTxtAsync());
                      
                     MessageBoxResult result = System.Windows.MessageBox.Show(
```
Comment (YuriZakitny) [inline DuplicatesFinder_v4/ViewModels/MainViewModel.cs:158]: not very correct.
Above you called async method without waiting.
This means that you will show messagebox before that method will complete

You need to somehow show this messagebox after task above will complete (maybe pass some lambda to called method)

**Review by YuriZakitny — COMMENTED** [review]
Very good.
Play more with async.

File: DuplicatesFinder_v4/ViewModels/MainViewModel.cs (hunk 1 above)
Comment (InnaMel) [reply DuplicatesFinder_v4/ViewModels/MainViewModel.cs:156]: Tnx. Fixed

File: DuplicatesFinder_v4/ViewModels/MainViewModel.cs (hunk 2 above)
Comment (InnaMel) [reply DuplicatesFinder_v4/ViewModels/MainViewModel.cs:158]: Tnx. Fixed. Used lambda.

**Pull request merged**